assembly/Bool.ts

```diff
@@ -51,9 +54,16 @@ export class Bool implements Codec {
      * @description Returns the string representation of the value
      */
     toString (): string {
-        return this.value.toString();
+        return this._value.toString();
+    }
+    
+    eq(other: Bool): bool {
+        return this._value == other._value;
```

Comment:
I think that `other._value` won't work. `_value` is private. You have to do `unwrap`?